pkg/providers/version/suite_test.go

```diff
@@ -70,9 +87,61 @@ var _ = AfterEach(func() {
 })
 
 var _ = Describe("Operator", func() {
-	It("should resolve Kubernetes Version via Describe Cluster", func() {
-		endpoint, err := awsEnv.VersionProvider.Get(ctx)
-		Expect(err).ToNot(HaveOccurred())
-		Expect(endpoint).To(Equal("1.30"))
+
+	Context("with EKS_CONTROL_PLANE=true", func() {
+		BeforeEach(func() {
+			versionProvider = version.NewDefaultProvider(env.KubernetesInterface, cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval), fakeEKSAPI)
+			os.Setenv("EKS_CONTROL_PLANE", "true")
```

Comment:
Once you migrate this over to options, you can set this directly in the options struct rather than updating the env var and reparsing options.